test/asynchronous/test_database.py

```diff
@@ -0,0 +1,778 @@
+        self.assertTrue(await db.validate_collection(db.test, True, True))
+
+    @async_client_context.require_version_min(4, 3, 3)
+    @async_client_context.require_no_standalone
+    async def test_validate_collection_background(self):
+        db = self.client.pymongo_test.with_options(write_concern=WriteConcern(w="majority"))
+        await db.test.insert_one({"dummy": "object"})
+        coll = db.test
+        self.assertTrue(await db.validate_collection(coll, background=False))
+        # The inMemory storage engine does not support background=True.
+        if async_client_context.storage_engine != "inMemory":
+            # background=True requires the collection exist in a checkpoint.
+            await self.client.admin.command("fsync")
+            self.assertTrue(await db.validate_collection(coll, background=True))
+            self.assertTrue(await db.validate_collection(coll, scandata=True, background=True))
+            # The server does not support background=True with full=True.
+            # Assert that we actually send the background option by checking
+            # that this combination fails.
+            with self.assertRaises(OperationFailure):
+                await db.validate_collection(coll, full=True, background=True)
+
+    async def test_command(self):
+        self.maxDiff = None
+        db = self.client.admin
+        first = await db.command("buildinfo")
+        second = await db.command({"buildinfo": 1})
+        third = await db.command("buildinfo", 1)
+        self.assertEqualReply(first, second)
+        self.assertEqualReply(second, third)
+
+    # We use 'aggregate' as our example command, since it's an easy way to
+    # retrieve a BSON regex from a collection using a command.
+    async def test_command_with_regex(self):
+        db = self.client.pymongo_test
+        await db.test.drop()
+        await db.test.insert_one({"r": re.compile(".*")})
+        await db.test.insert_one({"r": Regex(".*")})
+
+        result = await db.command("aggregate", "test", pipeline=[], cursor={})
+        for doc in result["cursor"]["firstBatch"]:
+            self.assertTrue(isinstance(doc["r"], Regex))
+
+    async def test_cursor_command(self):
+        db = self.client.pymongo_test
+        await db.test.drop()
+
+        docs = [{"_id": i, "doc": i} for i in range(3)]
+        await db.test.insert_many(docs)
+
+        cursor = await db.cursor_command("find", "test")
+
+        self.assertIsInstance(cursor, AsyncCommandCursor)
+
+        result_docs = await cursor.to_list()
+        self.assertEqual(docs, result_docs)
+
+    async def test_cursor_command_invalid(self):
+        with self.assertRaises(InvalidOperation):
+            await self.db.cursor_command("usersInfo", "test")
+
+    @async_client_context.require_no_fips
+    def test_password_digest(self):
+        with self.assertRaises(TypeError):
+            auth._password_digest(5)  # type: ignore
```

Comment:
We generally try to avoid using unspecified `# type: ignore`. You can ignore multiple typing rules at once like this: `# type: ignore[arg-type, call-arg]`